src/js/shot-generator/IK/objects/IkObjects/Ragdoll.js

```diff
@@ -0,0 +1,500 @@
+        if(this.hipsMouseDown)
+        {
+            let hipsTarget = this.hipsControlTarget.target;
+            let targetPosition = hipsTarget.position.clone();
+            let targetPos = hipsTarget.position.clone();
+            
+            targetPos.sub(this.objectTargetDiff);
+            this.clonedObject.position.copy(targetPos);
+            this.clonedObject.updateMatrix();
+            this.clonedObject.updateMatrixWorld(true);
+            
+            this.hips.parent.worldToLocal(targetPosition);
+            this.hips.position.copy(targetPosition);
+            this.hips.updateMatrix();
+            this.originalObject.position.copy(this.clonedObject.position);
+            this.updateCharPosition(this.clonedObject.position);
+        }
+    }
+
+    // Reintializes whole body and ik joints when character was changed
+    // Changing character height, head size will fire reinitialization
+    reinitialize()
+    {    
+        let chainObjects = this.chainObjects;
+        this.clonedObject.scale.copy(this.originalObject.scale);
+        this.clonedObject.position.copy(this.originalObject.position);
+        this.clonedObject.updateMatrixWorld(true);
+        for(let i = 0; i < chainObjects.length; i++)
+        {
+            let chain = chainObjects[i].chain;
+            chain.joints[chain.joints.length - 1].bone.getWorldPosition(chainObjects[i].controlTarget.target.position);
+
+            let poleConstraints = this.chainObjects[i].poleConstraint;
+            if(poleConstraints != null)
+            {
+                let targetPosition = new THREE.Vector3();
+                chain.joints[chain.joints.length - 2].bone.getWorldPosition(targetPosition);
+                let polePosition = poleConstraints.poleTarget.mesh.position;
+                poleConstraints.poleTarget.mesh.position.set(targetPosition.x + polePosition.x, targetPosition.y + polePosition.y, targetPosition.z + polePosition.z);
+                let poleTarget = poleConstraints.poleTarget;
+                this.calculatePoleTargetOffset(poleTarget, chain);
+                poleTarget.initialize(poleTarget.poleOffset);
+            }
+            chain.reinitializeJoints();
+        }
+        this.hips.getWorldPosition(this.hipsControlTarget.target.position);
+        this.calculteBackOffset();
+        this.ikSwitcher.applyToIk();
+        let hipsTarget = this.hipsControlTarget.target;
+        this.objectTargetDiff = new THREE.Vector3().subVectors(hipsTarget.position, this.originalObject.position);
+        this.setUpControlTargetsInitialPosition();
+    }
+
+    // Removes object and all it's meshes from scene
+    removeFromScene()
+     {
+         let scene = this.scene;
+         super.removeFromScene(scene);
+         this.controlTargetSelection.dispose();
+         this.chainObjects.forEach((chainObject)=>
+         {
+             if(chainObject.poleConstraint)
+             scene.remove(constraint.poleTarget.mesh);
```

Comment:
`constraint` is not defined. Error occurs here when changing the pose preset for a selected Character.

Reply:
Also get an error when attempting to select a Character preset.

Reply:
Haven't tested changing presets before. But the problem in remove method itself. Will fix it